Test for katello-tracer install through New Host UI

sambible opened the pull request:
First test using the New Hosts UI for the Tracer tab. Implementation is a bit hacky, not exactly familiar with how installing katello-tracer is done, so pieces from here and there were "appropriated" 

Requires https://github.com/SatelliteQE/airgun/pull/821

sambible commented:
trigger: test-robottelo
pytest: tests/foreman/ui/test_host.py -v -k 'test_positive_tracer_enable_reload'
airgun: 821

sambible commented:
trigger: test-robottelo
pytest: tests/foreman/ui/test_host.py -v -k 'test_positive_tracer_enable_reload'
airgun: 821

shubhamsg199 commented on tests/foreman/ui/test_host.py:
```diff
@@ -2597,3 +2613,40 @@ def test_positive_remove_role_in_new_ui(self):
+
+
+@pytest.mark.tier2
+@pytest.mark.rhel_ver_match('8')
```
Any specific reason we are testing for rhel8 and not rhel7 and rhel9 ?

sambible replied:
No, this was just an effort to limit the scope of the test to not take too long to run by being parametrized 5 or 6 more times. It shouldn't matter what version of Rhel the host is on, the test doesn't care, it just needs to be able to install katello-tracer

lhellebr replied:
In that case, I think we usually parametrize the tests with all the possible versions.

sambible replied:
ack, if I remove this decorator it will run on all possible versions, as well as FIPS versions.

lhellebr reviewed (changes requested):
Make sure the installation of Traces is done correctly, I suspect you forgot to put it in the fixture. Or the fixture's docstring just shouldn't say it enables it because it doesn't?

sambible commented:
This is a weird case, since I actually need it to not be installed on the host, just prepared for installation. This is because the workflow that I'm exercising actually uses rex to install the package, and then turn it on. Maybe I should make that more clear in the docstrings

sambible commented on tests/foreman/ui/test_host.py:
```diff
@@ -2597,3 +2613,40 @@ def test_positive_remove_role_in_new_ui(self):
+        title = session.host_new.get_tracer_title(prepared_host.hostname)
+        assert title == "Traces are being enabled"
+        # Waiting on traces to be enabled. Not optimal, but not sure how else to do it
+        sleep(15)
```
Not really sure what I'd wait_for on here, since the way the test functions is a job gets triggered in the background, and sometime after that job completes the page reloads.

lhellebr replied:
How about waiting for the condition you are asserting and fail if after certain time, the waiting hasn't finished?

sambible replied:
wait_for worked, thanks for the tip - hadn't used it before!

sambible commented:
trigger: test-robottelo
pytest: tests/foreman/ui/test_host.py -v -k 'test_positive_tracer_enable_reload'
airgun: 821

sambible commented:
Updated the docstring to reflect that this only sets up the host for katello-tracer installation, which is done during the test. Also updated it to run on all rhel7 and later hosts, as well as incorporated feedback around the wait, using wait_for now

vsedmik reviewed (approved):
Looks good to me

ogajduse reviewed (commented):
Great to see this addition! I feel this test is doing too less actions considering that it is spinning up a whole fleet of content hosts.
Keep in mind that spinning content hosts takes some time. I would rather see one robust UI test for the tracer rather than a bunch of tests testing just part of the component.
@sambible What are your future plans for the katello-tracer UI testing? Are you open to expanding this test?

lhellebr reviewed (approved):
Nice changes, LGTM

shubhamsg199 commented on tests/foreman/ui/test_host.py:
```diff
@@ -259,6 +259,21 @@ def remove_vm_on_delete(target_sat, setting_update):
 
 
+@pytest.fixture
+def prepared_host(rex_contenthost, target_sat):
```
Non-blocking, would suggest renaming this fixture as `prepared_host` seems something general.

lhellebr replied:
Yes.

shubhamsg199 reviewed (approved):
ACK

sambible commented:
@ogajduse Absolutely open to modifying/extending this test. I agree that this is a rather simple test, but that's mostly because this is responding to some small UI work in an area that hasn't been tested before. I can definitely see an extension of this test in the future to a more general end to end test, but for now I'm just trying to get enough coverage to satisfy the requirements for a 6.13 release

sambible commented on tests/foreman/ui/test_host.py:
```diff
@@ -2597,3 +2612,47 @@ def test_positive_remove_role_in_new_ui(self):
+        title = session.host_new.get_tracer_title(prepared_host.hostname)
+        assert title == "Traces are being enabled"
+        # Waiting on traces to be enabled. Not optimal, but not sure how else to do it
+        wait_for(
+            lambda: session.host_new.get_tracer_title(prepared_host.hostname)
+            != "Traces are being enabled",
+            timeout=1800,
+            delay=5,
+            silent_failure=True,
+            handle_exception=True,
+        )
+        title = session.host_new.get_tracer_title(prepared_host.hostname)
```
It does return the lambda value, unfortunately that value is a boolean, since I'm doing a comparison. No clean way I can think of to have this function properly and also return the pre comparison value.

ogajduse replied:
If you look closely at my suggestion, I have moved the comparison to `fail_condition`. But this is non-blocking and could be eventually addressed in #11112.

JacobCallahan commented:
@sambible if you raise an issue for the future enhancements, we'll merge it 😉

sambible commented:
https://github.com/SatelliteQE/robottelo/issues/11112 - done :)

ogajduse commented:
I have put two bullet points to #11112 as recommendations for future improvement. I am merging the PR to unblock testing of the katello-tracer component.

sambible commented:
going to rebase/merge based on @ogajduse comments.

JacobCallahan commented:
@sambible we discourage self-merges. Can you also look into the failed cherry-pick?

ogajduse commented:
@sambible on @sambible's defense, I have suggested self-merge as there were a bunch of approvals and we want that coverage to be included ASAP.

ogajduse commented:
@JacobCallahan, on @sambible's defense, that was me who suggested self-merge as there were a bunch of approvals and we want that coverage to be included ASAP.
